swift/trainers/cpo_trainer.py

```diff
@@ -1,28 +1,30 @@
-from typing import Any, Dict, Union
+from typing import Any, Dict, List, Tuple, Union
 
 import torch
 from torch import nn
-from transformers import PreTrainedModel, trainer
+from transformers import PreTrainedModel
 from trl import CPOTrainer as HFCPOTrainer
 
 from swift.llm.utils.template import Template
 from swift.llm.utils.utils import sort_by_max_length
 from swift.utils import get_logger
-from .callback import DefaultFlowCallbackNew, PrinterCallbackNew, ProgressCallbackNew
 from .mixin import PushToMsHubMixin, SwiftMixin
-from .utils import build_tokenized_answer, concat_template
+from .utils import build_tokenized_answer, patch_trl
 
 logger = get_logger()
+patch_trl()
 
 
 class CPOTrainer(PushToMsHubMixin, SwiftMixin, HFCPOTrainer):
 
     def __init__(self, *args, template: Template, test_oom_error=False, **kwargs):
         self.template = template
         kwargs.pop('gamma', None)
+        is_multimodal = kwargs.pop('is_multimodal')
```

Comment:
Is there a better way to confirm if it's a vision model? Because multimodal includes audio and video as well.